### PR TITLE
Factions: basic foundation

### PR DESCRIPTION
**Pull request by fromage-fraser**
Basic declarations and management commands for an experimental
"factions" concept.

Factions are randomly generated allegiances of mobiles with their own
personalities, abilities and buffs. They do not persist across restarts.

- Define a basic lexicon of words for generating random names.
  (Note: name generation is still pretty janky!)

- Basic faction properties: alignment, demeanour.

- Generate factions at server start up.

- 'factions' wiz command to list faction info.

### CODE REVIEW

File: server/src/lexicon.c [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+        { "sanctum",    LEXICON_FILTER_GOOD | LEXICON_FILTER_LAWFUL },
+
+        { "law",        LEXICON_FILTER_LAWFUL },
+        { "book",       LEXICON_FILTER_GENERAL | LEXICON_FILTER_LAWFUL },
+        { "grimoire",   LEXICON_FILTER_EVIL | LEXICON_FILTER_LAWFUL },
+};
+
+/*
+ * Main function for selecting a random entry from an entry list, with optional filtering.
+ * Supply a filter value of zero to consider all entries.
```
Comment (henry-fraser) [inline server/src/lexicon.c:257]: Ugh, bug: I actually need to check for zero in the code...

File: server/player/g/Gezhp [hunk 2]
```diff
@@ -13,7 +13,7 @@ ClnLvl      0
 Lvl         106
 Trst        0
 Wizbt       0
-Playd       5902
+Playd       10367
```
Comment (shadenz1) [inline server/player/g/Gezhp:16]: I'm thinking you can discard changes to Gezhp :)

**Review by shadenz1 — DISMISSED** [review]
Looks good to me

File: server/src/lexicon.c (hunk 1 above)
Comment (shadenz1) [reply server/src/lexicon.c:257]: Is this handled before hit merge?

Comment (fromage-fraser) [reply server/src/lexicon.c:257]: Zero is now handled correctly!

**Comment by fromage-fraser** [comment]
Closing PR for now. Will keep branch and think about whether this will get done in reasonable time or not!

**Pull request closed without merging**